### PR TITLE
Added user service/model for registration

### LINKED ISSUE #57: Setup DB
### Make models for: 

- [x] user

- [x] posts

- [x] dialogs

- [x] chats
- [x] online users

### PR DESCRIPTION
**Pull request by grok-rs**
**Changes**
1. Added service for getting user information by Id api/user/:id
2. I decided to transfer post service, which was separated to user service, that looks more consistently
3. Added service for login, register. 
Main logic of login includes:
- get email from google service (token);
- check this mail in DB, if exists we return user info, else return null and route to register 
4. Added services for posts: getByUserId and createPost

### CODE REVIEW

**Review by Pr1sm0 — APPROVED** [review]
LGTM

File: backend/src/controller/index.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import express from 'express';
 import usersRoute from './user';
+import postRoute from './post';
```
Comment (oleksandrstarov) [inline backend/src/controller/index.ts]: please align namings usersRoute but postRoute seems inconsistent and then routes are user and posts

File: backend/src/data/models/user.ts [hunk 2]
```diff
@@ -1,7 +1,53 @@
...
+    type: String,
+    required: true,
+    trim: true
+  },
+  lastName: {
+    type: String,
+    required: true,
+    trim: true
+  },
+  sex: Number,
```
Comment (oleksandrstarov) [inline backend/src/data/models/user.ts]: really?

File: backend/src/data/repository/post.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  async createPost(newPost: DTO.IPost) {
+    const post = new Post(newPost);
+    await post.save(function (err, post) {
+      if (err) return console.error(err);
+      console.log(post);
+    });
+    return post;
+  }
```
Comment (oleksandrstarov) [inline backend/src/data/repository/post.ts]: potential issue here - we creating post and return this directly. Not the saved copy which comes as second param from callback function

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**